src/devices/tuya.ts

```diff
@@ -20,8 +20,100 @@ const fzZosung = zosung.fzZosung;
 const tzZosung = zosung.tzZosung;
 const ez = zosung.presetsZosung;
 
+function isNumber(value: number) {
+    return !isNaN(value) && isFinite(value);
+
+    // return /^-?\d+$/.test(value);
+}
+
+function formattedToRaw(key: string, input: string, number: number, payload: number[], meta: Tz.Meta) {
+    const items = input.trim().split(/\s+/);
+    let message = '';
+    let OK: boolean = true;
+
+    if (items.length != number) {
+        // meta.state.message = 'Wrong number of items for '+ key +' :' + items.length;
+        // throw new Error('Wrong number of items for '+ key +' :' + items.length);
+        message = 'Wrong number of items for '+ key +' :' + items.length;
+    }
+    if (OK) {
+        for (let i = 0; i < number; i++) {
+            const hourTemperature = items[i].split('/');
+            const hourMinute = hourTemperature[0].split(':', 2);
+            const hour = parseInt(hourMinute[0]);
+            const minute = parseInt(hourMinute[1]);
+            const temperature = parseInt(hourTemperature[1]);
+
+            if (!isNumber(hour) || !isNumber(temperature) || !isNumber(minute) || hour < 0 || hour >= 24 ||
+                minute < 0 || minute >= 60 || temperature < 5 || temperature >= 35) {
+                message = 'Invalid hour, minute or temperature (5<t<35) in '+ key +' of: `' + items[i]+'`; Format is `hh:m/cc.c` or `hh:mm/cc.c°C`';
+
+                // throw new Error('Invalid hour, minute or temperature in '+ key +' of: `' + items[i]+"`; Format is `hh:m/cc.c` or `hh:mm/cc.c°C`");
+                OK= false;
+                break;
+            }
+            const temperature10 =parseInt(hourTemperature[1])*10;
+
+            payload.push(hour);
+            payload.push(minute);
+            payload.push((temperature10 >> 8) & 0xFF);
+            payload.push( temperature10 & 0xFF);
+        }
+    }
+    return message;
+}
 
 const tzLocal = {
+
+    // Since some keys needed, are not defined in datapoints, the <datapoints> is extended here with needed datapoints
+
+    ZWT198_datapoints:
```

Comment:
Please add the missing keys to the `tuya.datapoints` converter

Reply:
Changed key to an existing key in tuya.datapoints